application/common/components/Emailer.php

```diff
@@ -694,24 +706,36 @@ public function sendPasswordExpiredEmails()
             return;
         }
 
+        $logData = [
+            'action' => 'send password expired notices',
+            'status' => 'starting',
+        ];
+
+        $users = User::getActiveUnlockedUsers();
+
+        $this->logger->info(array_merge($logData, [
+            'active_users' => count($users)
+        ]));
+
         $numEmailsSent = 0;
-        $users = User::findAll(['active' => 'yes', 'locked' => 'no', ]);
         foreach ($users as $user) {
             /** @var Password $userPassword */
             $userPassword = $user->currentPassword;
-            if ($userPassword
-                && strtotime($userPassword->getExpiresOn()) < time()
-                && ! $this->hasReceivedMessageRecently($user->id, EmailLog::MESSAGE_TYPE_PASSWORD_EXPIRED)
-            ) {
-                $this->sendMessageTo(EmailLog::MESSAGE_TYPE_PASSWORD_EXPIRED, $user);
-                $numEmailsSent++;
+            if ($userPassword) {
+                $passwordExpiry = strtotime($userPassword->getExpiresOn());
+                if ($passwordExpiry < time()
+                    && $passwordExpiry > strtotime(self::FIFTEEN_DAYS_AGO)
```

Comment:
You don't need to change it at this point, but just to clarify, I was thinking more along the lines of a constant that describes what this is for, e.g. PASSWORD_EXPIRY_OUTDATED_CUTOFF (not putting too much thought into it)